sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs

```diff
@@ -0,0 +1,233 @@
+
+            var locationOptions = new List<LocationOptionsQuery>
+            {
+                new LocationOptionsQuery
+                {
+                    LabelId = "state",
+                    OptionsValue = "WA"
+                },
+                new LocationOptionsQuery
+                {
+                    LabelId = "city",
+                    OptionsValue = "NOAM-US-WA-SE"
+                }
+            };
+
+            var areaCodes = await client.GetAllAreaCodesAsync("selection", countryCode, phonePlanId, locationOptions);
+
+            Assert.IsNotNull(areaCodes.Value.PrimaryAreaCodes);
+            Assert.IsTrue(areaCodes.Value.PrimaryAreaCodes.Count > 0);
+        }
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("en-US")]
+        [AsyncOnly]
+        public async Task CreateReservationErrorState(string? locale)
+        {
+            var client = CreateClient();
+            var countryCode = "US";
+
+            var pageablePhonePlanGroups = client.GetPhonePlanGroupsAsync(countryCode, locale);
+            var phonePlanGroups = await pageablePhonePlanGroups.ToEnumerableAsync().ConfigureAwait(false);
+
+            string phonePlanGroupId = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.TollFree).PhonePlanGroupId;
+            var pageablePhonePlans = client.GetPhonePlansAsync(countryCode, phonePlanGroupId, locale);
+            var phonePlan = (await pageablePhonePlans.ToEnumerableAsync()).First();
+            var areaCode = phonePlan.AreaCodes.First();
+
+            var reservationOptions = new CreateReservationOptions("My reservation", "my description", new[] { phonePlan.PhonePlanId }, areaCode);
+            reservationOptions.Quantity = 1;
+            var reservationOperation = await client.StartReservationAsync(reservationOptions);
+
+            try
+            {
+                await reservationOperation.WaitForCompletionAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Assert.AreEqual("Reservation has failed.", ex.Message);
+                return;
+            }
+
+            Assert.Fail("WaitForCompletionAsync shoul have throun exception.");
+
+            Assert.IsNotNull(reservationOperation);
+            Assert.IsTrue(reservationOperation.HasCompleted);
+            Assert.IsTrue(reservationOperation.HasValue);
+
+            var reservation = reservationOperation.Value;
+            Assert.IsNotNull(reservation);
+
+            Assert.AreEqual(ReservationStatus.Reserved, reservation.Status);
+            Assert.AreEqual(areaCode, reservation.AreaCode);
+            Assert.IsNull(reservation.ErrorCode);
+            Assert.AreEqual(1, reservation.PhoneNumbers?.Count);
+        }
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("en-US")]
+        [AsyncOnly]
+        public async Task CreateReservation(string? locale)
```

Comment:
The recordings for `CreateReservationErrorState` don't show any non-200 status codes. Maybe `CreateReservationErrorState` isn't completely implemented and just an early copy paste? :-)

Reply:
We do not have any non 200 status codes to take care of. Those statuses are handled in the generated code.
When reservation is failing we receive a reservation object with 200 http status, but payload has a Reservation.Status == error and that case is tested in CreateReservationErrorState